easybuild/tools/toolchain/toolchain.py

```diff
@@ -95,6 +99,10 @@ def __init__(self, name=None, version=None, mns=None):
 
         self.vars = None
 
+        self.add_class_constants_to_restore([])  # make sure self.CLASS_CONSTANTS_TO_RESTORE is initialised
```

Comment:
Defining a class constant as a value of a mutable type is a problem, especially if you update it (see add_class_constants_to_restore).

I need to make sure a new list gets created for every toolchain class (`Ictce` vs `Goalf` vs `Intel`, etc.).

That doesn't happen if you set it to `[]` in `Toolchain` (outside of `__init__`).